### ISSUE TITLE
MD005 and MD007 rules conflict, causing infinite loop

### ISSUE THREAD

**Issue by alies-dev** [rvben/rumdl#287]
# Bug: MD005 and MD007 rules conflict, causing infinite loop

## Environment
- rumdl version: 0.0.214
- OS: macOS (Darwin 24.6.0)

## Description
When running `rumdl fmt` on a markdown file with nested lists, the linter fails to converge after 100 iterations due to conflicting requirements between MD005 and MD007 rules.

## Minimal Reproducible Example

Create a file `test.md` with this content:

```markdown
# Test

1. First item
1. Second item
  - nested bullet
  - another bullet
1. Third item
   - nested under third
   - another nested
1. Fourth item
  1. sub-numbered
     - deep nested
```

Run:
```bash
rumdl fmt test.md
```

## Expected Behavior
The linter should either:
1. Fix the file to a consistent state, or
2. Report a clear error about conflicting configuration

## Actual Behavior
The linter outputs:
```
Warning: Failed to converge after 100 iterations.
This likely indicates a bug in rumdl.
Rule codes: MD005, MD007
Please report at: https://github.com/rvben/rumdl/issues/new
```

The rules conflict:
- MD005 reports: "Expected indentation of 2 spaces, found 3"
- MD007 reports: "Expected 3 spaces for indent depth 1, found 2"

The linter alternates between these two states indefinitely.

## Workaround
Currently none - the affected lines cannot be auto-fixed.

**Comment by rvben** [comment]
Thanks for the detailed bug report, @alies-dev!

Fixed in v0.0.216.

The oscillation was caused by MD005 and MD007 fighting over indent widths when ordered and unordered lists were mixed. The fix groups list items by their marker type (ordered vs unordered) and calculates consistent indentation within each group.

Your example now formats without oscillation.
